dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/DataAccessManager.kt

```diff
@@ -0,0 +1,179 @@
+import org.dataland.datalandcommunitymanager.model.dataRequest.AccessStatus
+import org.dataland.datalandcommunitymanager.model.dataRequest.RequestStatus
+import org.dataland.datalandcommunitymanager.repositories.DataRequestRepository
+import org.dataland.datalandcommunitymanager.utils.AccessRequestLogger
+import org.dataland.datalandcommunitymanager.utils.DataRequestLogger
+import org.dataland.datalandcommunitymanager.utils.DataRequestProcessingUtils
+import org.dataland.keycloakAdapter.auth.DatalandAuthentication
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.Instant
+
+/**
+ * The DataAccessManager contains methods for the data access logic.
+ */
+@Service
+class DataAccessManager(
+    @Autowired private val dataRequestRepository: DataRequestRepository,
+    @Autowired private val dataRequestLogger: DataRequestLogger,
+    @Autowired private val dataRequestProcessingUtils: DataRequestProcessingUtils,
+    @Autowired private val accessRequestLogger: AccessRequestLogger,
+) {
+
+    /**
+     * This method finds all DataRequestsEntity for a specified dataset that have a specific accessStatus.
+     * @param companyId the companyId for which the access status should be checked
+     * @param reportingPeriod the reportingPeriod for which the access status should be checked
+     * @param dataType the framework dataType for which the access status should be checked
+     * @param userId the userId for which the access status should be checked
+     */
+    fun hasAccessToPrivateDataset(
+        companyId: String,
+        reportingPeriod: String,
+        dataType: DataTypeEnum,
+        userId: String,
+    ): Boolean {
+        val hasAccess = dataRequestRepository
+            .findByUserIdAndDatalandCompanyIdAndDataTypeAndReportingPeriod(
+                userId, companyId, dataType.name, reportingPeriod,
+            )?.any { it.accessStatus == AccessStatus.Granted } ?: false
+
+        if (hasAccess) {
+            accessRequestLogger.logMessageForCheckingIfUserHasAccessToDataset(
+                companyId,
+                dataType,
+                reportingPeriod,
+            )
+        }
+        return hasAccess
+    }
+
+    /**
+     * This method checks if the requesting user has granted access to the information for a specific company,
+     * reporting period and data type.
+     * If there is no access or no data set an exception is thrown.
+     * @param companyId the companyId for which the access status should be checked
+     * @param reportingPeriod the reportingPeriod for which the access status should be checked
+     * @param dataType the framework dataType for which the access status should be checked
+     * @param userId the userId for which the access status should be checked
+     */
+    @Transactional
+    fun headAccessToDataset(
```

Comment:
weird method name. I understand that there's a HEAD call behind, but imho this is not a meaningful method name


also: "this method checks if the requesting" sounds like the method should return a boolean

Reply:
Transcactional not necessary here

Reply:
Transactional removed and name changed to hasAccessToDataset